File: src/main/java/org/adempiere/install/IDatabaseConfig.java
package org.adempiere.install;

/**
 *
 * @author hengsin
 *
 */
public interface IDatabaseConfig {

	/**
	 * Get real database name from native connection profile name
	 * return from discoverDatabases
	 * @param nativeConnectioName
	 * @return Database name
	 */
	public String getDatabaseName(String nativeConnectioName);


	/**
	 * 	Discover Databases.
	 * 	To be overwritten by database configs
	 *	@param selected selected database
	 *	@return array of databases
	 */
	public String[] discoverDatabases(String selected);


	/**
	 *  Get Database Name
	 *  @return database short name
	 */
	public String getName();
}

File: src/main/kotlin/org/adempiere/install/IDatabaseConfig.kt
package org.adempiere.install

/**
 *
 * @author hengsin
 */
interface IDatabaseConfig {

    /**
     * Get Database Name
     * @return database short name
     */
    val name: String

    /**
     * Get real database name from native connection profile name
     * return from discoverDatabases
     * @param nativeConnectioName
     * @return Database name
     */
    fun getDatabaseName(nativeConnectioName: String): String

    /**
     * Discover Databases.
     * To be overwritten by database configs
     * @param selected selected database
     * @return array of databases
     */
    fun discoverDatabases(selected: String): Array<String>
}